File: 백준/Silver/3273. 두 수의 합/두 수의 합.py
n = int(input()) # 수열의 크기
numbers = list(map(int, input().split()))   # 수열에 포함되는 수
x = int(input())    # 두 수의 합

numbers.sort()
result = 0
s = 0
e = n-1
while s < e:
    if numbers[s] + numbers[e] == x:
        result += 1
        s += 1
        e -= 1
    elif numbers[s] + numbers[e] < x:
        s += 1
    else:
        e -= 1

print(result)

File: 백준/Silver/3273. 두 수의 합/두 수의 합.js
let fs = require("fs");
let input = fs.readFileSync("/dev/stdin").toString().split("\n");

let n = Number(input[0]);
let arr = input[1]
  .split(" ")
  .map(Number)
  .sort((a, b) => a - b);
let x = Number(input[2]);
let result = 0;

for (let i = 0; i < n; i++) {
  let j = i + 1;
  while (j < n) {
    if (arr[i] + arr[j] == x) {
      result++;
      break;
    } else if (arr[i] + arr[j] < x) {
      j++;
    } else break;
  }
}
console.log(result);
